src/Perspex.Controls/Window.cs

```diff
@@ -325,3 +325,16 @@ private void EnsureInitialized()
         }
     }
 }
+
+namespace Perspex
+{
+    public static class WindowApplicationExtensions
+    {
+        public static void RunWithMainWindow<WindowT>(this Application app) where WindowT : Perspex.Controls.Window, new()
```

Comment:
Can we use `TWindow` for the generic type? https://msdn.microsoft.com/en-us/library/0zk36dx2.aspx